core/server/api/v2/utils/validators/input/slugs.js

```diff
@@ -0,0 +1,16 @@
+const common = require('../../../../../lib/common');
+const models = require('../../../../../models');
+
+module.exports = {
+    all(apiConfig, frame) {
+        const allowedTypes = {
+            post: models.Post,
+            tag: models.Tag,
+            user: models.User,
+            app: models.App
+        };
```

Comment:
Don't like the fact that whole `allowedTypes` gets duplicated here as it is also [here](https://github.com/TryGhost/Ghost/pull/9978/files#diff-2314dc08a53031b6341a0c803447a669R15), might need a better way to avoid duplicating it 🤔

Reply:
Yeah def will make it look better, but its still kind of duplicating the possible `allowedTypes`, as in they'd need changes in 2 places. Would it make sense(maybe future?) to extend and configure `frame` in a way that it can contain some meta stuff like this so it can be passed down?

Reply:
Oh i think you can already do that

```
validation
  options
     type: ['post'...]
```

Reply:
👍 This is perfect, have updated to actually just use allowedTypes once with this pattern.